django/chat/responses.py

```diff
@@ -420,25 +422,39 @@ async def add_files_to_library():
         )
 
     # Summarize mode
-    if chat.options.qa_mode == "summarize":
-        # Use summarization on each of the documents
+    if chat.options.qa_mode in ["summarize", "summarize_combined"]:
         if not filter_documents:
             filter_documents = Document.objects.filter(
                 data_source__library=chat.options.qa_library
             )
         document_titles = [document.name for document in filter_documents]
-        summary_responses = [
-            llm.tree_summarize(
-                context=document.extracted_text,
+        if chat.options.qa_mode == "summarize_combined":
+            # Combine all documents into one text, including the titles
+            combined_documents = "\n\n---\n\n".join(
```

Comment:
Does this *always* guarantee that the AI will properly delineate documents? I'm imagining a situation where one of the docs contains this same pattern of line breaks and bold text, which would presumably be falsely interpreted as another document separation.

Reply:
Fair. I should probably change this to XML style.